Source/MenubarFlag.spoon/init.lua

```diff
@@ -0,0 +1,202 @@
+--- === MenubarFlag ===
+---
+--- Color the menubar according to the current keyboard layout
+---
+--- Download: [https://github.com/Hammerspoon/Spoons/raw/master/Spoons/MenubarFlag.spoon.zip](https://github.com/Hammerspoon/Spoons/raw/master/Spoons/MenubarFlag.spoon.zip)
+---
+--- Functionality inspired by [ShowyEdge](https://pqrs.org/osx/ShowyEdge/index.html.en)
+
+local obj={}
+obj.__index = obj
+
+local draw = require "hs.drawing"
+local col = draw.color.x11
+
+-- Metadata
+obj.name = "MenubarFlag"
+obj.version = "0.1"
+obj.author = "Diego Zamboni <diego@zzamboni.org>"
+obj.homepage = "https://github.com/Hammerspoon/Spoons"
+obj.license = "MIT - https://opensource.org/licenses/MIT"
+
+--- MenubarFlag.allScreens
+--- Variable
+--- Boolean to specify whether the indicators should be shown on all monitors or just the current one. Defaults to `true`
+obj.allScreens = true
+
+--- MenubarFlag.indicatorHeight
+--- Variable
+--- Number to specify the height of the indicator. Specify 0.0-1.0 to specify a percentage of the height of the menu bar, larger values indicate a fixed height in pixels. Defaults to 1.0
+obj.indicatorHeight = 1.0
+
+--- MenubarFlag.indicatorAlpha
+--- Variable
+--- Number to specify the indicator transparency (0.0 - invisible; 1.0 - fully opaque). Defaults to 0.3
+obj.indicatorAlpha = 0.3
+
+--- MenubarFlag.indicatorInAllSpaces
+--- Variable
+--- Boolean to specify whether the indicator should be shown in all spaces (this includes full-screen mode). Defaults to `true`
+obj.indicatorInAllSpaces = true
+
+--- MenubarFlag.colors
+--- Variable
+--- Table that contains the configuration of indicator colors
+---
+--- The table below indicates the colors to use for a given keyboard
+--- layout. The index is the name of the layout as it appears in the
+--- input source menu. The value of each indicator is a table made of
+--- an arbitrary number of segments, which will be distributed evenly
+--- across the width of the screen. Each segment must be a valid
+--- `hs.drawing.color` specification (most commonly, you should just
+--- use the named colors from within the tables). If a layout is not
+--- found, then the indicators are removed when that layout is active.
+---
+--- Inidicator specs can be static flag-like:
+--- ```
+---   Spanish = {col.green, col.white, col.red},
+---   German = {col.black, col.red, col.yellow},
+--- ```
+--- or complex, programmatically-generated:
+--- ```
+--- ["U.S."] = (
```

Comment:
Because the lack of a "U.S." entry means "no indicator" - however I have added an explicit empty spec which has the same effect, just for clarity.